Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -89,6 +89,25 @@ struct EventProcessParams
         { MidIOHelper.STARPOWER_DRUM_FILL_4, ProcessNoteOnEventAsDrumFill },
     };
 
+        // These dictionaries map the text of a MIDI text event to a specific function that processes them
+        static readonly Dictionary<string, EventProcessFn> GuitarTextEventToProcessFnMap = new Dictionary<string, EventProcessFn>()
+    {
+    };
+
+        static readonly Dictionary<string, EventProcessFn> GhlGuitarTextEventToProcessFnMap = new Dictionary<string, EventProcessFn>()
+    {
+    };
+
+        static readonly Dictionary<string, EventProcessFn> DrumsTextEventToProcessFnMap = new Dictionary<string, EventProcessFn>()
+    {
+        { MidIOHelper.CHART_DYNAMICS_TEXT, (in EventProcessParams eventProcessParams) => {
+            BuildDrumsMidiNoteNumberToProcessFnDict(enableVelocity: true);
+        }},
+        { MidIOHelper.CHART_DYNAMICS_TEXT_BRACKET, (in EventProcessParams eventProcessParams) => {
+            BuildDrumsMidiNoteNumberToProcessFnDict(enableVelocity: true);
+        }}
```

Comment:
Honestly I'm not sure how I didn't catch that it would persist between songs lol

Looking into a better implementation, it's not gonna be a single-commit change but it's not gonna be a full-on rewrite of anything.

Reply:
New implementation works great! Took a number of changes to get there, but future parsing features that require on-the-fly changing of how things are parsed (such as guitar `[ENHANCED_OPENS]`) should be able to be handled a little more sanely.